DAO.sol

```diff
@@ -890,28 +941,40 @@ contract DAO is DAOInterface, Token, TokenCreation {
     }
 
     function numberOfProposals() constant returns (uint _numberOfProposals) {
-        // Don't count index 0. It's used by isBlocked() and exists from start
+        // Don't count index 0. It's used by getOrModifyBlocked() and exists from start
         return proposals.length - 1;
     }
 
     function getNewDAOAddress(uint _proposalID) constant returns (address _newDAO) {
         return proposals[_proposalID].splitData[0].newDAO;
     }
 
-    function isBlocked(address _account) internal returns (bool) {
+    function getOrModifyBlocked(address _account) internal returns (bool) {
         if (blocked[_account] == 0)
             return false;
         Proposal p = proposals[blocked[_account]];
-        if (now > p.votingDeadline) {
+        if (p.open) {
             blocked[_account] = 0;
             return false;
         } else {
             return true;
         }
     }
 
+    function getOrModifySplitBlocked(address _account) internal returns (bool) {
+        if (splitBlocked[_account] == 0)
+            return false;
+        Proposal p = proposals[splitBlocked[_account]];
+        if (p.open) {
```

Comment:
`splitExecutionPeriod` is 27 days as can be seen [here](https://github.com/slockit/DAO/blob/master/DAO.sol#L37). Are you sure you mean that? Perhaps you mean the `splitGracePeriod` from [this](https://github.com/slockit/DAO/pull/246/files#diff-b30f9d3be2a832949bdd72e23956c130R44) PR.

If it's the latter `p.open` stays true until `executeProposal()` is succesfull, right? It's kind of the same check, only relying on someone having succesfully called `executeProposal()`. In the case of a faulty proposal which always throws, someone can always call `executeProposal()` to have it closed after the `waitingPeriod()` is over.